src/main/java/hu/oe/nik/szfmv/automatedcar/model/ParkingSpace.java

```diff
@@ -2,9 +2,49 @@
 
 import hu.oe.nik.szfmv.automatedcar.model.interfaces.IBackground;
 import hu.oe.nik.szfmv.automatedcar.model.interfaces.IStatic;
+import hu.oe.nik.szfmv.automatedcar.model.utility.Consts;
+
+import javax.xml.bind.Unmarshaller;
+import java.awt.Rectangle;
 
 /**
  * Parkolóhely alaposztály.
  */
 public class ParkingSpace extends WorldObject implements IStatic, IBackground {
+    private final int LEFT_BORDER = 4;
+    private final int RIGHT_BORDER = 14;
+    private final int TOP_BORDER = 10;
+    private final short BOTTOM_BORDER = 4;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void afterUnmarshal(Unmarshaller u, Object parent) {
+        super.afterUnmarshal(u, parent);
+        this.position.setZ(1);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initShape() {
+        switch (this.imageFileName) {
+            case Consts.RES_IDENTIFIER_PARKINGSPACE_PARALLEL:
+                this.initParallelShape();
+                break;
+            case Consts.RES_IDENTIFIER_PARKINGSPACE_90:
+                // TODO: implement parking space 90
+                throw new UnsupportedOperationException("Not implemented yet");
```

Comment:
Fogalmam sincs, hogy erre miért nem szól be a fordító. Javaban nem kellene kezelni minden exceptiont kezelni azonnal, vagy jelölni a függvényben?

Reply:
A RuntimeExceptionök jelzéséért, kezeléséért, nem szól a fordító, így nem szükséges kezelned, mert jellemzően ezek olyan hibák, amire nem számítasz, nem direkt dobod. Ezért javasoltam, hogy kerüljön külön try-catchbe. Itt pl simán előfordulhat olyan, hogy egy xml-ből létrehozunk egy objektumot ezzel a képpel.